src/frontend/src/flows/addDevice/welcomeView/index.ts

```diff
@@ -3,10 +3,16 @@ import { registerTentativeDevice } from "./registerTentativeDevice";
 import { mkAnchorInput } from "../../../components/anchorInput";
 import { Connection } from "../../../utils/iiConnection";
 
-const pageContent = (connection: Connection, userNumber?: bigint) => {
+const addRemoteDeviceTemplate = ({
+  onContinue,
+  userNumber,
+}: {
+  onContinue: (userNumber: bigint) => void;
+  userNumber?: bigint;
+}) => {
   const anchorInput = mkAnchorInput({
     userNumber,
-    onSubmit: (userNumber) => registerTentativeDevice(userNumber, connection),
+    onSubmit: onContinue,
```

Comment:
ultra nth: you can rename when you deconstruct

```
const addRemoteDeviceTemplate = ({
  onContinue: onSubmit,
  userNumber,
}: {
  const anchorInput = mkAnchorInput({
    userNumber,
    onSubmit
```

Reply:
Thanks! added